File: CORE/Enums/OrderStatus.cs
namespace CORE.Enums
{
    /// <summary>
    /// OrderStatus - defines the possible states of an order throughout its lifecycle
    /// </summary>
    public enum OrderStatus
    {
        /// <summary>Order has been created but not yet processed</summary>
        Pending,

        /// <summary>Order has been confirmed and is being prepared for shipment</summary>
        Confirmed,

        /// <summary>Order has been shipped and is in transit</summary>
        Shipped,

        /// <summary>Order has been successfully delivered</summary>
        Delivered,

        /// <summary>Order has been cancelled by customer or system</summary>
        Cancelled,
    }
}

File: WEB/ecommerce-web/src/types/enums/orderStatus.ts
/**
 * Order status enum - represents the state of an order
 */
export const OrderStatus = {
  Pending: 'Pending',
  Shipped: 'Shipped',
  Delivered: 'Delivered',
  Cancelled: 'Cancelled'
} as const;

export type OrderStatus = typeof OrderStatus[keyof typeof OrderStatus];